### PR TITLE
Add `Octo issue develop` command

### PR DESCRIPTION
**Pull request by wd60622**
Adds `issue develop` command. 

If on issue buffer, the current issue is checked out and the branch is created.
If not, then a picker is shown to select an issue to checkout and create a branch.

NOTE: Only implemented for the telescope picker

Related to #443

### CODE REVIEW

File: lua/octo/commands.lua [hunk 1]
```diff
@@ -62,6 +62,18 @@ function M.setup()
...
+      develop = function(repo, ...)
+        local bufnr = vim.api.nvim_get_current_buf()
+        local buffer = octo_buffers[bufnr]
+
+        if not buffer then
+          local opts = M.process_varargs(repo, ...)
+          picker.issues(opts, true)
+          return
+        end
+
```
Comment (wd60622) [inline lua/octo/commands.lua:74]: Need to add clause for not issue buffer

**Review by wd60622 — COMMENTED** [review]
Still need to work through any documentation, mappings, or configuration

Any thoughts here?

**Comment by pwntester** [comment]
Sorry, was not able to take a look. Will do it today

**Comment by wd60622** [comment]
> Sorry, was not able to take a look. Will do it today

Thanks @pwntester 

Note: I adapted the `M.process_varargs` which allows for repo as argument. This potentially shouldn't work for develop as the issue numbers from another repo wouldn't make sense

File: lua/octo/commands.lua [hunk 2]
```diff
@@ -62,6 +62,17 @@ function M.setup()
...
       end,
+      develop = function(repo, ...)
+        local bufnr = vim.api.nvim_get_current_buf()
+        local buffer = octo_buffers[bufnr]
+
+        if buffer and buffer.kind and buffer.kind == "issue" then
+          utils.develop_issue(buffer.node.number)
+        else
+          local opts = M.process_varargs(repo, ...)
+          picker.issues(opts, true)
```
Comment (pwntester) [inline lua/octo/commands.lua:73]: Do you expect to pass any of the arguments handled by varargs to `Octo issue develop`? If so, this should be ok

**Review by pwntester — APPROVED** [review]

File: lua/octo/commands.lua (hunk 2 above)
Comment (wd60622) [reply lua/octo/commands.lua:73]: Do not expect to pass any args here. Was using the functions that already exist in order to get the current repo (when repo is not passed). But could just duplicate the logic instead. 

This works as is but could cause some confusion if digging into this.

What are your thoughts?

**Comment by wd60622** [comment]
TODO: 

- [ ] Add command to the README

Any other items?

File: lua/octo/commands.lua (hunk 2 above)
Comment (pwntester) [reply lua/octo/commands.lua:73]: I think we can pass `repo` and `develop` as part of the opts so there should not be any need for handling varargs or passing develop as a separate argument.

Comment (wd60622) [reply lua/octo/commands.lua:73]: Are you suggesting a change then?

Comment (wd60622) [reply lua/octo/commands.lua:73]: The issue with passing `repo` as an argument is that the location of that repo locally is not known. So, only the current repo is truely supported.

Comment (wd60622) [reply lua/octo/commands.lua:73]: I can create an issue to clean up this in the future

**Pull request merged**